### PR TITLE
Fix listener of color options and input blocker for side bar

### PR DESCRIPTION
**Pull request by optimize-2**

### CODE REVIEW

File: src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java [hunk 1]
```diff
@@ -79,16 +79,23 @@ public void draw(long vg, int x, int y, InputHandler inputHandler) {
...
                 colorSelector = new ColorSelector(finalColor, inputHandler.mouseX(), inputHandler.mouseY(), allowAlpha, inputHandler);
                 OneConfigGui.INSTANCE.initColorSelector(colorSelector);
             }, 1);
         }
         if (OneConfigGui.INSTANCE.currentColorSelector != colorSelector) open = false;
-        else if (open) color = (OneConfigGui.INSTANCE.getColor());
-        setColor(color);
+        else if (open) {
+            try {
+                if (!OneConfigGui.INSTANCE.getColor().toJavaColor().equals(((OneColor) field.get(parent)).toJavaColor()))  {
```
Comment (nextdayy) [inline src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java]: making both of these java colors is redundant or should be redundant

File: src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java [hunk 2]
```diff
@@ -79,16 +79,23 @@ public void draw(long vg, int x, int y, InputHandler inputHandler) {
...
+                    ((OneColor) field.get(parent)).setFromOneColor(OneConfigGui.INSTANCE.getColor());
```
Comment (nextdayy) [inline src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java]: i'm going to assume you have tested this

**Review by nextdayy — CHANGES_REQUESTED** [review]

File: src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java (hunk 2 above)
Comment (optimize-2) [reply src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java]: > i'm going to assume you have tested this

I've really tested this i promise

Comment (nextdayy) [reply src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java]: okay, can you still change the other issue i pointed out.

File: src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java (hunk 1 above)
Comment (optimize-2) [reply src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java]: oops this was for testing
i was locating the bug so i wrote this

**Comment by optimize-2** [comment]
it's done i think

**Comment by optimize-2** [comment]
wait i forget to push something

**Review by Wyvest — DISMISSED** [review]

File: src/main/java/cc/polyfrost/oneconfig/config/elements/BasicOption.java [hunk 3]
```diff
@@ -92,6 +92,10 @@ public BasicOption(Field field, Object parent, String name, String description,
     protected void set(Object object) throws IllegalAccessException {
         if (field == null) return;
         field.set(parent, object);
+        this.triggerListeners();
+    }
+
+    public void triggerListeners() {
```
Comment (dynpatch) [inline src/main/java/cc/polyfrost/oneconfig/config/elements/BasicOption.java]: Run `:apiDump`. Also this should probably be `protected`, not `public`

**Review by dynpatch — DISMISSED** [review]

File: src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java [hunk 4]
```diff
@@ -79,16 +79,23 @@ public void draw(long vg, int x, int y, InputHandler inputHandler) {
...
-        else if (open) color = (OneConfigGui.INSTANCE.getColor());
-        setColor(color);
+        else if (open) {
+            try {
+                if (!OneConfigGui.INSTANCE.getColor().equals((OneColor) field.get(parent)))  {
+                    ((OneColor) field.get(parent)).setFromOneColor(OneConfigGui.INSTANCE.getColor());
+                    this.triggerListeners();
+                }
+            } catch (IllegalAccessException ignore) {
+            }
```
Comment (nextdayy) [inline src/main/java/cc/polyfrost/oneconfig/gui/elements/config/ConfigColorElement.java]: please move this code to replace the `setColor` method.

**Review by nextdayy — DISMISSED** [review]

**Review by Wyvest — APPROVED** [review]

**Pull request merged**